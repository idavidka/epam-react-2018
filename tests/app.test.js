import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Search from '../lib/search';

jest.mock('../lib/search.jsx', () => 'Search');

describe('App', () => {
    it('should render html structure', () => {
        const search = renderer.create(<Search title="Test" placeholder="Test" button="Button" />).toJSON();

        console.log(search);
        expect(search).toMatchSnapshot();
    });

    it('should do something', () => {
        const eventMock = {
            target: {
                value: {}
            }
        };

        const search = shallow(<Search title="Test" placeholder="Test" button="Button" />);

        console.log(search.debug());
        search.find('input').simulate('change', eventMock);
    });
});