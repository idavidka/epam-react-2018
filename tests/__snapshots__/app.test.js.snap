// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render html structure 1`] = `
<Search
  button="Button"
  placeholder="Test"
  title="Test"
/>
`;
